### DISCUSSION TITLE
react top level import for module federation demo

### DISCUSSION THREAD

**Discussion by zhtyytg** [web-infra-dev/rsbuild#1785] [Q&A] [answered]
I'm quite new for rsbuild, react and module federation.
Using `pnpm create rsbuild` to create a new react sub module at a pnpm monorepo. Then add module federation config as a host sharing react & react-dom ( without any other config). 
![image](https://github.com/web-infra-dev/rsbuild/assets/42018740/e3bd3d23-f21d-4b8b-b275-fb4d2da07bc5)

Then I check the rsbuild example. The difference between them is then example wrapped the React app root with `import('./bootstrap').cath(xxxx)`. 
Does rsbuild do some transfer when compiling `import` if configured module federation？
I want to know if this configuration is the expected behavior recommended by the official

**Comment by chenjiahan** [comment, marked as answer]
The `import('./bootstrap').cath(xxxx)` is an asynchronous boundary, it is recommended to use by the offical module federation guide, see:

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/ecbd644c-5b54-4e09-a6b9-a338deca7a5c)

- https://webpack.js.org/concepts/module-federation/#troubleshooting
- https://module-federation.io/docs/en/mf-docs/0.2/shared-api/
